### PR TITLE
industries (so far) and error handling

### PR DESCRIPTION
**Pull request by ethn**

### CODE REVIEW

File: mod/metrics/set/abstract/value.rb [hunk 1]
```diff
@@ -28,68 +32,10 @@ def company_id
...
-
-event :mark_as_imported, before: :finalize_action, when: :import_act? do
-  @current_action.comment = "imported"
-end
-
-# def answer_imported?
-#   ActManager.act_card.type_id.in? [AnswerImportFileID, RelationshipAnswerImportFileID]
+# def record
+#   name.parts[0..-3].join "+"
+# end
```
Comment (xithan) [inline mod/metrics/set/abstract/value.rb]: seems like good api to me. Why is it commented?

File: mod/metrics/set/type/free_text_value.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+include_set Abstract::Value
+
+format :html do
+  view :editor do
+    text_field :content, class: "d0-card-content"
+  end
+
+  def pretty_value
+    @pretty_value ||= card.value
+  end
```
Comment (xithan) [inline mod/metrics/set/type/free_text_value.rb]: that could be the default in `Abstract::Value`

File: mod/metrics/set/type/metric/creation.rb [hunk 3]
```diff
@@ -78,7 +78,7 @@ def create_value_args args
     type_id: type_id,
     "+value" => {
       content: args[:value],
-      type_id: (args[:value].is_a?(Integer) ? NumberID : PhraseID)
+      type_code: value_cardtype_code
```
Comment (xithan) [inline mod/metrics/set/type/metric/creation.rb]: dot missing ?!

File: mod/isic/spec/set/type/isic_class_value_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# -*- encoding : utf-8 -*-
+
+RSpec.describe Card::Set::Type::IsicClassValue do
+
```
Comment (xithan) [inline mod/isic/spec/set/type/isic_class_value_spec.rb]: would be good to have at least one test that checks that pulling the options from json is working

File: mod/metrics/set/type/metric/answers.rb [hunk 5]
```diff
@@ -70,16 +71,23 @@ def value_type_code
   value_type_card.item_cards.first&.codename || default_value_type_code
 end
 
+def value_cardtype_code
+  :"#{value_type_code}_value"
+end
```
Comment (xithan) [inline mod/metrics/set/type/metric/answers.rb]: can we document somewhere what all these variants of compositions of "value", "code", and "type" mean and what kind of value they hold?

**Review by xithan — CHANGES_REQUESTED** [review]
"If it's not tested, it doesn't work"

File: mod/metrics/set/type/metric/creation.rb (hunk 3 above)
Comment (xithan) [reply mod/metrics/set/type/metric/creation.rb]: probably correct this way

File: mod/isic/spec/set/type/isic_class_value_spec.rb (hunk 4 above)
Comment (ethn) [reply mod/isic/spec/set/type/isic_class_value_spec.rb]: agree. all that code is now in decko, though.  will write it there.

File: mod/metrics/set/abstract/value.rb (hunk 1 above)
Comment (ethn) [reply mod/metrics/set/abstract/value.rb]: I commented because it wasnt used.  but I restored it.

**Comment by ethn** [comment]
The JSON options are now entirely handled (and lightly tested) in Decko.

**Review by xithan — APPROVED** [review]

**Pull request merged**